pkg/control/sidecarcontrol/util_hotupgrade.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2020 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sidecarcontrol
+
+import (
+	"encoding/json"
+	"fmt"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+)
+
+const (
+	// record which hot upgrade container is working currently
+	SidecarSetWorkingHotUpgradeContainer = "kruise.io/sidecarset-working-hotupgrade-container"
+
+	// hotUpgrade container name suffix
+	hotUpgradeNameSuffix1 = "-1"
+	hotUpgradeNameSuffix2 = "-2"
+
+	// sidecar container version in container env(SIDECARSET_VERSION)
+	SidecarSetVersionEnvKey = "SIDECARSET_VERSION"
+	// sidecar container version in container env(SIDECARSET_VERSION_ALT)
```

Comment:
container version env  in the other sidecar container of the same hotupgrade sidecar